### ISSUE TITLE
Godot Game Engine TorchSharp Environment

### ISSUE THREAD

**Issue by GeorgeS2019** [dotnet/TorchSharp#1032]
This is WIP to figure out how to configure TorchSharp environment in Godot .NET6/7 environment
If anyone has experience, please share.

![image](https://github.com/dotnet/TorchSharp/assets/49812372/c0780697-0ec9-4c21-9150-27022122ddb1)

**Comment by GeorgeS2019** [comment]
I sort the problem.  Create issue for future Godot users coming to TorchSharp

**Comment by NiklasGustafsson** [comment]
As mentioned high up on the README page:

We recommend using one of the 'bundled' packages: TorchSharp-cpu, TorchSharp-cuda-windows, or TorchSharp-cuda-linux. They will pull in the right libtorch backends.

**Comment by GeorgeS2019** [comment]
@NiklasGustafsson 
I tried, so as @ChengYen-Tang 
It does not work with Godot. It will take some time to investigate

**Comment by michieal** [comment]
okay, this kinda begs the question -- Godot builds for different platforms. Which one does one use (the package) for something like that?

**Comment by NiklasGustafsson** [comment]
The question of platform is (unfortunately) tied to libtorch as well as TorchSharp. 

libtorch supports Windows x64, Linux x64, Linux ARM64, MacOS x64, MacOS M1. You can supposedly build it from source for other platforms, per my understanding. Of these, we haven't tested TorchSharp with Linux ARM or MacOS M1 (because I don't have the hardware for either).

**Comment by GeorgeS2019** [comment]
@NiklasGustafsson 
Should I write a formal request to ask Microsoft allocate these hardware for you ☺️

**Comment by NiklasGustafsson** [comment]
> @NiklasGustafsson Should I write a formal request to ask Microsoft allocate these hardware for you ☺️

No, thanks! :-)

**Comment by GeorgeS2019** [comment]
@NiklasGustafsson 
I recently wrote a long email to someone why TorchSharp is important to Deep AI for 3D game engine   like Godot with MIT license and ranked 8th in the entire GitHub universe.

**Comment by michieal** [comment]
> The question of platform is (unfortunately) tied to libtorch as well as TorchSharp.
> 
> libtorch supports Windows x64, Linux x64, Linux ARM64, MacOS x64, MacOS M1. You can supposedly build it from source for other platforms, per my understanding. Of these, we haven't tested TorchSharp with Linux ARM or MacOS M1 (because I don't have the hardware for either).

So, different projects for different build platforms. 
And, I understand the whole "don't have the hardware" aspect.

**Comment by NiklasGustafsson** [comment]
Not quite -- the TorchSharp-cpu package pulls in Windows/Linux/MacOS x64, so for those, you don't have to do anything special. If you need a CUDA backend, it's different.

**Comment by michieal** [comment]
okay, cool.
And, I wouldn't do the CUDA one as it's not guaranteed that the end user will have an Nvidia card, or ROCm drivers that would support it. That, and depending on the user's video card to run something like that, with it having full graphics, is pushing it. So... I would just use the CPU one. 

Also, I saw that llama.cpp (github project) has a PR (at the time I saw it) that made use of the nmap? c++ command that allows the  model files to be mapped as memory space (like a dedicated swap file) and I was wondering if TorchSharp has something like that or could work with something created in c# to do that? 

(Probably should have asked that on a different topic/issue. my bad)
